src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,134 @@
+
+    public void postBegin() {
+        boolean processedOnce = false;
+        for (EntityRef entity : entityManager.getEntitiesWith(WorldComponent.class)) {
+            if (!processedOnce) {
+                delayManager.addPeriodicAction(entity, BODY_TEMPERATURE_UPDATE_ACTION_ID, 0, CHECK_INTERVAL);
+                processedOnce = true;
+            } else {
+                logger.warn("More than one entity with WorldComponent found");
+            }
+        }
+        // The world is the entity here which shall have a Body Temperature Update Periodic Action.
+    }
+
+    @ReceiveEvent
+    public void onTemperatureUpdate(PeriodicActionTriggeredEvent event, EntityRef world) {
+        if (event.getActionId().equals(BODY_TEMPERATURE_UPDATE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(AliveCharacterComponent.class,
+                    BodyTemperatureComponent.class, LocationComponent.class)) {
+                LocationComponent location = entity.getComponent(LocationComponent.class);
+                BodyTemperatureComponent bodyTemperature = entity.getComponent(BodyTemperatureComponent.class);
+                float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition());
+                float envHumidity = climateConditionsSystem.getHumidity(location.getLocalPosition());
+                float deltaTemp =
+                        ((((envTemperature - (envHumidity / 10)) - bodyTemperature.current) / 100000) * CHECK_INTERVAL);
+
+                //Send event for other systems to modify change in body temperature.
+                if(deltaTemp >= 0) {
+                    AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(deltaTemp);
+                    entity.send(affectBodyTemperatureEvent);
+                    deltaTemp = affectBodyTemperatureEvent.getResultValue();
+                } else {
+                    AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(-1 * deltaTemp);
+                    entity.send(affectBodyTemperatureEvent);
+                    deltaTemp = -1 * affectBodyTemperatureEvent.getResultValue();
+                }
+                //Check for change in body temperature levels.
+                BodyTemperatureLevel before = lookupLevel(bodyTemperature.current);
+                float oldValue = bodyTemperature.current;
+                //Update current body temperature.
+                bodyTemperature.current = bodyTemperature.current + deltaTemp;
+                entity.saveComponent(bodyTemperature);
+                float newValue = bodyTemperature.current;
+                BodyTemperatureLevel after = lookupLevel(bodyTemperature.current);
+
+                if (before != after) {
+                    entity.send(new BodyTemperatureLevelChangedEvent(before, after));
+                    bodyTemperature.currentLevel = after;
+                }
```

Comment:
IMO this check should be the result of receiving the `BodyTemperatureValueChangedEvent`
(like "oh the value changed, this means the level could've changed, let's check that)